spring-cloud-stream-binders/spring-cloud-stream-binder-kafka/src/main/java/org/springframework/cloud/stream/binder/kafka/KafkaProducerProperties.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.binder.kafka;
+
+import org.springframework.cloud.stream.binder.ProducerProperties;
+import org.springframework.integration.kafka.support.ProducerMetadata;
+
+/**
+ * @author Marius Bogoevici
+ */
+public class KafkaProducerProperties extends ProducerProperties {
+
+	private int bufferSize = 16384;
+
+	private ProducerMetadata.CompressionType compressionCodec = ProducerMetadata.CompressionType.none;
+
+	private boolean sync = false;
+
+	private KafkaMessageChannelBinder.Mode mode = KafkaMessageChannelBinder.Mode.embeddedHeaders;
+
+	public int getBufferSize() {
+		return bufferSize;
+	}
+
+	public void setBufferSize(int bufferSize) {
+		this.bufferSize = bufferSize;
+	}
+
+	public ProducerMetadata.CompressionType getCompressionCodec() {
+		return compressionCodec;
+	}
+
+	public void setCompressionCodec(ProducerMetadata.CompressionType compressionCodec) {
```

Comment:
even though I do like "codec" better, I wonder if the name of the property should match the enum type: `setCompressionType(..)`